hathor/event/event_manager.py

```diff
@@ -0,0 +1,55 @@
+import time
+
+from structlog import get_logger
+
+from hathor.event.base_event import BaseEvent
+from hathor.event.storage import EventStorage
+from hathor.pubsub import EventArguments, HathorEvents, PubSubManager
+
+logger = get_logger()
+
+
+class EventManager:
+    def __init__(self, event_storage: EventStorage, peer_id: str):
+        self.event_storage = event_storage
+        last_event = event_storage.get_last_event()
+        self.next_event_index = 1 if last_event is None else last_event.id+1
+        self.peer_id = peer_id
+        self.log = logger.new()
+
+    def subscribe(self, pubsub: PubSubManager) -> None:
+        """ Subscribe to defined events for the pubsub received
+        """
+        self.log.error(pubsub)
+        events = [
+            HathorEvents.NETWORK_NEW_TX_ACCEPTED,
+            HathorEvents.NETWORK_NEW_TX_VOIDED,
+            HathorEvents.NETWORK_BEST_BLOCK_FOUND,
+            HathorEvents.NETWORK_ORPHAN_BLOCK_FOUND,
+            HathorEvents.LOAD_STARTED,
+            HathorEvents.LOAD_FINISHED,
+            HathorEvents.REORG_STARTED,
+            HathorEvents.REORG_FINISHED,
+            HathorEvents.TX_METADATA_CHANGED,
+            HathorEvents.BLOCK_METADATA_CHANGED
+        ]
+
+        for event in events:
+            pubsub.subscribe(event, self._persist_event)
+
+    def _persist_event(self, event: HathorEvents, args: EventArguments) -> None:
+        event_data = args.__dict__['event']
+        event_to_store = BaseEvent(
+            id=self.next_event_index,
+            peer_id=self.peer_id,
+            timestamp=int(time.time()),  # Current epoch
+            type=event.name,
+            data=event_data,
+            group_id=0,  # TODO
+        )
+        try:
+            self.event_storage.save_event(event_to_store)
+        except Exception:
+            self.log.error('Error while saving event', event=event_to_store)
```

Comment:
Should we kill the node when it happens? I mean, it could cause integrations to lose events.

Reply:
I agree, I think we should let the error bubble up and just not catch it here.

Reply:
Done. I removed the try/except.

Reply:
Great!